config/src/main/java/com/typesafe/config/DefaultConfigLoadingStrategy.java

```diff
@@ -12,51 +14,11 @@
 public class DefaultConfigLoadingStrategy implements ConfigLoadingStrategy {
     @Override
     public Config parseApplicationConfig(ConfigParseOptions parseOptions) {
-        ClassLoader loader = parseOptions.getClassLoader();
-        if (loader == null)
-            throw new ConfigException.BugOrBroken(
-                    "ClassLoader should have been set here; bug in ConfigFactory. "
-                            + "(You can probably work around this bug by passing in a class loader or calling currentThread().setContextClassLoader() though.)");
-
-        int specified = 0;
-
-        // override application.conf with config.file, config.resource,
-        // config.url if requested.
-        String resource = System.getProperty("config.resource");
-        if (resource != null)
-            specified += 1;
-        String file = System.getProperty("config.file");
-        if (file != null)
-            specified += 1;
-        String url = System.getProperty("config.url");
-        if (url != null)
-            specified += 1;
-
-        if (specified == 0) {
+        Config overrideConfig = ConfigFactory.parseApplicationOverride(parseOptions);
+        if (overrideConfig.isEmpty()) {
```

Comment:
with optional it could be `orElseGet(() -> ConfigFactory.parseResourcesAnySyntax("application", parseOptions))` maybe